src/contracts/core/StakeRootCompendium.sol

```diff
@@ -234,226 +202,155 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
-        uint256 numStrategiesBefore = operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length();
 
-        for (uint256 i = 0; i < strategies.length; i++) {
-            require(
-                operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].remove(address(strategies[i])),
-                "StakeRootCompendium.removeStrategiesAndMultipliers: strategy not found"
+    function _addStrategiesAndMultipliers(
+        IAVSDirectory.OperatorSet memory operatorSet,
+        StrategyAndMultiplier[] calldata strategiesAndMultipliers
+    ) internal {
+        // update the deposit balance for the operator set whenever number of strategies is changed
+        _updateDepositInfo(operatorSet);
+        uint256 numStrategiesBefore = operatorSetToStrategyAndMultipliers[msg.sender][operatorSet.operatorSetId].length();
+        require(
+            numStrategiesBefore + strategiesAndMultipliers.length <= MAX_NUM_STRATEGIES, 
+            "StakeRootCompendium.setStrategiesAndMultipliers: too many strategies"
+        );
+        // set the strategies and multipliers for the operator set
+        for (uint256 i = 0; i < strategiesAndMultipliers.length; i++) {
+            operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].set(
+                address(strategiesAndMultipliers[i].strategy), uint256(strategiesAndMultipliers[i].multiplier)
             );
         }
-
-        _updateTotals(numStrategiesBefore, operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length());
+        _updateTotalStrategies(numStrategiesBefore, operatorSet);
     }
 
-    function _removeOperatorSet(IAVSDirectory.OperatorSet memory operatorSet) internal {
-        if(operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId].length() != 0) {
-            bytes32[] memory strategyBytes = operatorSetToStrategyAndMultipliers[operatorSet.avs][operatorSet.operatorSetId]._inner._keys.values();
-            IStrategy[] memory strategies = new IStrategy[](strategyBytes.length);
-            // todo: better way to do this?
-            for (uint256 i = 0; i < strategyBytes.length; i++) {
-                strategies[i] = IStrategy(address(uint160(uint256(strategyBytes[i]))));
-            }
-
-            // remove the strategies and multipliers for the operator set
-            _removeStrategiesAndMultipliers(
-                operatorSet,
-                strategies
-            );
-        }
-
-        IAVSDirectory.OperatorSet memory substituteOperatorSet = operatorSets[operatorSets.length - 1];
-        uint224 operatorSetIndex = operatorSetToIndex[operatorSet.avs][operatorSet.operatorSetId].latest();
-        operatorSets[operatorSetIndex] = substituteOperatorSet;
-        operatorSets.pop();
+    function _updateTotalStrategies(uint256 numStrategiesBefore, IAVSDirectory.OperatorSet memory operatorSet) internal {
```

Comment:
rename back to totalCharge because we need to call this when charges change

Reply:
will do